### PR TITLE
US 642597: Add FIPS flag for pega infinity

### PR DESCRIPTION
**Pull request by GaneshKatta95**

### CODE REVIEW

File: charts/pega/templates/pega-environment-config.yaml [hunk 1]
```diff
@@ -185,12 +185,15 @@ data:
...
   {{ if (eq (include "isHzHighlySecureCryptoModeEnabled" .) "true") }}
-  HIGHLY_SECURE_CRYPTO_MODE_ENABLED: "true"
   HZ_SSL_ALGO: "PKIX"
   {{- else }}
   HZ_SSL_ALGO: "SunX509"
   {{- end }}
 {{- end }}
+{{- end }}
+
+{{ if (eq (include "isPegaHighlySecureCryptoModeEnabled" .) "true") }}
```
Comment (Saurabh-16) [inline charts/pega/templates/pega-environment-config.yaml:195]: Can we cover this functionality as part of go test in terratest module?

File: terratest/src/test/pega/pega-environment-config_test.go [hunk 2]
```diff
@@ -77,6 +77,38 @@ func TestPegaEnvironmentConfigJDBCTimeouts(t *testing.T) {
...
+					"global.actions.execute":               operation,
+					"global.highlySecureCryptoModeEnabled": "false",
+				},
+			}
+
+			yamlContent := RenderTemplate(t, options, helmChartPath, []string{"templates/pega-environment-config.yaml"})
+			VerifyEnvNotPresent(t, yamlContent, "HIGHLY_SECURE_CRYPTO_MODE_ENABLED")
+			
+			options.SetValues["global.highlySecureCryptoModeEnabled"] = "true"
+			yamlContent := RenderTemplate(t, options, helmChartPath, []string{"templates/pega-environment-config.yaml"})
```
Comment (Saurabh-16) [inline terratest/src/test/pega/pega-environment-config_test.go:105]: Its failing here. Probably you need to replace `:=` with `=`

Comment (Saurabh-16) [reply terratest/src/test/pega/pega-environment-config_test.go:105]: https://github.com/pegasystems/pega-helm-charts/actions/runs/12482818554/job/34837635816?pr=858

Comment (GaneshKatta95) [reply terratest/src/test/pega/pega-environment-config_test.go]: Updated the test

**Review by Saurabh-16 — APPROVED** [review]

**Review by arvasrikanth — APPROVED** [review]

**Pull request merged**